src/main/java/org/jabref/model/entry/field/FieldFactory.java

```diff
@@ -210,7 +206,6 @@ private static Set<Field> getAllFields() {
         fields.addAll(EnumSet.allOf(InternalField.class));
         fields.addAll(EnumSet.allOf(SpecialField.class));
         fields.addAll(EnumSet.allOf(StandardField.class));
-        fields.removeIf(field -> field instanceof UserSpecificCommentField);
```

Comment:
Why are user specific fields re added?

Reply:
The code in this method does not add any UserSpecificCommentFields. That means, they are NOT added. That means, it is NOT necessary to remove them. Thus, the to remove them is not required. Thefere, I removed the code.

Thus, they are NOT re-added, because the method did not add them at all.

Here a screenshot of the whole method:

![image](https://github.com/JabRef/jabref/assets/1366654/b106943c-03c2-4599-b632-df885c543483)

No "UserSpecificCommentField".

Reply:
But aren't they apart of InternalField?

Reply:
`InternalField` is something like citaton key, all-text-fields, markedentry, etc.

It is an `enum`:

```java
public enum InternalField implements Field {
```

See https://github.com/JabRef/jabref/blob/948b84b5b578fce291adeb07a3436249ee2c13bf/src/main/java/org/jabref/model/entry/field/InternalField.java#L10

The `UserSpecificCommentField`  is a class implementing Field:

```java
public class UserSpecificCommentField implements Field {
```

---

Side note: Did you know that by pressing dot (`.`) while having any PR opened opens a visual studio code web instance with full code access?